### ISSUE TITLE
Why making a one-file template?

### ISSUE THREAD

**Issue by giampierosalvi** [martstr/beamerthementnu#13]
Thank you for the template, it is very useful. However, it is not clear to me why making a one-file template. Beamer has a neat way to define all the style specific configurations in a theme file with the following advantages:
* the only thing you need in the main tex file is to issue the \usetheme command, hiding all the complications away,
* the rest of the presentation would use standard beamer commands,
* it would be easy to change the style from the NTNU template to another without changing the main file,
* the template could be uploaded to overleaf as one of the official NTNU templates, making it even easier to use.
The current implementation, in my opinion, makes the main file unnecessarily complicated, scaring away people who are not so familiar with LaTeX, but would like to give it a try.

Is there any reason not to use a theme file to define the template? Thank you!

**Comment by kellertuer** [comment]
Hi,
at first glance the template looks quite nice. I can try to provide the template as a style file the next days, such that it can be included by setting the LaTeX Beamer theme.

**Comment by kellertuer** [comment]
Ah, that was faster than I thought, see #14 or the new slides covering some ideas/options https://github.com/kellertuer/beamerthementnu/blob/master/beamerntnu2020/ntnu-beamer-documentation.pdf

**Comment by kellertuer** [comment]
I had a little LaTeX Beamer fun today, so my PR now also covers (see documentation in the previous post) three styles (was too lazy to build the circles style for now)

* plain https://github.com/kellertuer/beamerthementnu/blob/master/beamerntnu2020/ntnu-beamer-example_plain.pdf
* stripe https://github.com/kellertuer/beamerthementnu/blob/master/beamerntnu2020/ntnu-beamer-example_stripe.pdf
* stripebottom https://github.com/kellertuer/beamerthementnu/blob/master/beamerntnu2020/ntnu-beamer-example_stripebottom.pdf

all these can be set when loading the theme, e.g. `\usetheme[style=stripe]{NTNU}` and one can switch the slogan using the option `slogan=norsk` to the Norwegian slogan/logo.

**Comment by VeroniMattia** [comment]
Dear all, thanks for your enquiry. When I was asked to create a tex version of the ppt template, I was also asked to make it as simple as possible. After a discussion with the department for communication, we agreed upon a stand-alone file as the easiest approach for new Tex users. I am very happy to help with the development of a style file on the side.

**Comment by giampierosalvi** [comment]
Hi @kellertuer,
thank you very much for the improvements! It looks very good. Next time I do a presentation I will have more feedback for sure :)

@VeroniMattia I am not sure how a complicated stand-alone file would be easier for new TeX users. The beauty of LaTeX is to separate structure from style. New users should only need to learn few commands related to the structure of the document and not a maze of commands mixing content and style. I have been using LaTeX for 20+ years, and I found the old template very hard to work with. That's however just my humble opinion.

Thank you both for providing these very useful resources.

**Comment by VeroniMattia** [comment]
@giampierosalvi - I feel you. I don't have that much of experience, but I also prefer using .sty files. The problem is that we have plenty of users that are not comfortable with them, and I got many emails confirming this.
I think that having both is definitely the best option, in order to satisfy the needs of all types of users :) 
@kellertuer - You have done an amazing job, and I'm sure that there will be only small adjustments, if any. I guess you can become a co-responsible, if not the main one to maintain this repository. Let me know if you might be interested :)

**Comment by kellertuer** [comment]
Thanks for the discussion.

Can you help me in one point for the feedback you get? If they have a single-file tex bit still have to copy the logos, why would it not be ok to also copy the sty-files?
My goal would be to provide a thorough documentation and an example one can directly start with – one that is just a few lines and easy to understand (a single-file-one would now be 500+ lines?)
Also keeping both sty-files _and_ a single file one, would mean to always copy code (manually?!) and have to copies of the code – I am not available for such a repository. I will symlink the style files into my latex tree, so I need style files.

Thanks for the positive feedback - let me know about the adjustments; one can also try to get the circle one included (if we get the graphics) – for now I was too lazy for that.

Conceringn co-responsibility – with the above mentioned copy-approach I would not be available. I just joined NTNU in March (for long term though), but sure in general I would be available. It would – however – be reasonable to move the repo to a community repo (found a community ntnu-latex or something). That would make the URL independent of the maintainer. 
On Such an Org, one could also do the following:
We have a repository with style files (the _main_ one, and I would be available to co-maintain that) and a repository with a one-file variant that someone (not me) maintains and adapts whenever the main one is changed. A clever solution could do this copy-into-one-file with a shell-script and that could also commit&push – but I am not a shell script expert, so another reason I would not be available for single-file solutions.

**Comment by giampierosalvi** [comment]
If I may contribute my two cents to the discussion: I was TeX master at KTH (in Sweden) for many years before moving to Norway, and I developed similar templates there. One thing that I learned is that it is essential to simplify as much as possible, because the official graphic profile is likely to be updated regularly at NTNU (as it was at KTH) and the templates are going to need updates too. I, therefore, strongly recommend to only keep a single version.

@VeroniMattia: I would be interested to better understand the problems those users encountered with the .sty files. If the templates are done correctly, you will never have to touch the .sty files, only to copy them in the current directory. I suspect that the implementation they were complaining about required them to modify the .sty file. In this case I totally agree that an inexperienced TeX user would easily get lost. But again, in a proper implementation, the only file the user should be worried about is the .tex file. On top of this, the .tex file will be significantly simpler than the current implementation, so I only see a simplification from the user's point of view.

To sum up, I strongly recommend to keep a version with separate .sty files. I haven't tested Ronny's implementation yet, but as long as it works by just changing the .tex file, it should be a huge improvement both for experienced and inexperienced LaTeX users.

**Comment by kellertuer** [comment]
Thanks for the comment – you are completely right, no user that just aims to use the theme should have to look into the style file. All things that a user might want to change can be changed by options. These options are well-documented in a documentation (itself a presentation to illustrate also how to use them).
One aim is, that a `example_presentation_StyleA` (maybe one for each file) is as simple as possible to directly start, and simple here means also as few lines as possible to not get confused.

**Comment by VeroniMattia** [comment]
For the moment, I have merged the two branches and added a sty folder, keeping the standalone files. I'll meet with kellertuer on Friday to discuss the next steps. I am very happy with dropping the previous version and leaving only the sty folder, and I am happy as well to contribute to it and to its documentation, but not as the main maintainer 😄

**Comment by kellertuer** [comment]
Currently, though, the style file approach also has the single-file examples included.

All in all I am too new to NTNU to take over main maintenance, I think; especially I am not sure which obligations come with being the main maintainer. But we can see about that on Friday.

**Comment by kellertuer** [comment]
this is resolved basically since the style files are available, general discussions now in #16.
